WooCommerce/Classes/ViewRelated/Orders/Collect Payments/CollectOrderPaymentUseCase.swift

```diff
@@ -454,87 +455,102 @@ private extension CollectOrderPaymentUseCase {
-                                               }
-                                           }, dismissCompletion: {
-                                               onCompletion(.failure(error))
-                                           })
-        )
+                                               }, dismissCompletion: {
+                                                   onCompletion(.failure(error))
+                                               })
+            )
+        }
     }
 
     private func presentRetryWithoutRestartingError(error: Error,
                                                     paymentAlerts: any CardReaderTransactionAlertsProviding<AlertPresenter.AlertDetails>,
                                                     paymentGatewayAccount: PaymentGatewayAccount,
                                                     onCompletion: @escaping (Result<CardPresentCapturedPaymentData, Error>) -> ()) {
-        alertsPresenter.present(
-            viewModel: paymentAlerts.error(
-                error: error,
-                tryAgain: { [weak self] in
-                    guard let self = self else { return }
-                    self.checkOrderIsStillEligibleForPayment(alertProvider: paymentAlerts, onPaymentCompletion: onCompletion) { result in
-                        switch result {
-                        case .failure(let error):
-                            return self.checkThenHandlePaymentFailureAndRetryPayment(error,
-                                                                                     alertProvider: paymentAlerts,
-                                                                                     paymentGatewayAccount: paymentGatewayAccount,
-                                                                                     onCompletion: onCompletion)
-                        case .success:
-                            self.paymentOrchestrator.retryPayment(for: self.order) { [weak self] result in
-                                guard let self = self else { return }
-                                switch result {
-                                case .success(let capturedPaymentData):
-                                    self.handleSuccessfulPayment(capturedPaymentData: capturedPaymentData)
-                                    onCompletion(.success(capturedPaymentData))
-                                case .failure(CardReaderServiceError.paymentMethodCollection(.commandCancelled(let cancellationSource))):
-                                    switch cancellationSource {
-                                    case .reader:
-                                        self.handlePaymentCancellationFromReader(alertProvider: paymentAlerts)
-                                    default:
-                                        self.handlePaymentCancellation(from: .other)
+        receiptStateForFailedPayment(error: error, order: order) { [weak self] receiptState in
```

Comment:
The review view shows a lot of changes, but I only wrapped it in `receiptStateForFailedPayment(error: error, order: order) { [weak self] receiptState in`